File: Lowest Common Ancestor of a Binary Search Tree - Leetcode 235/Lowest Common Ancestor of a Binary Search Tree - Leetcode 235.cpp
class Solution {
public:
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        TreeNode* lca = root;

        while (root != nullptr) {
            if (root->val < p->val && root->val < q->val) {
                root = root->right;
            } else if (root->val > p->val && root->val > q->val) {
                root = root->left;
            } else {
                lca = root;
                break;
            }
        }

        return lca;
    }
};

File: Lowest Common Ancestor of a Binary Tree - Leetcode 235/Lowest Common Ancestor of a Binary Search Tree - Leetcode 235.java
public class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        TreeNode lca = root;

        while (root != null) {
            if (root.val < p.val && root.val < q.val) {
                root = root.right;
            } else if (root.val > p.val && root.val > q.val) {
                root = root.left;
            } else {
                lca = root;
                break;
            }
        }

        return lca;
    }
}
